lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -83,19 +83,43 @@ def cache_sql(sql, *args)
       super(_host_pool_current_database.to_s + "/" + sql, *args)
     end
   end
+
+  module PoolConfigPatch
+    def pool
+      ActiveSupport::ForkTracker.check!
+
+      @pool || synchronize { @pool ||= ActiveRecordHostPool::PoolProxy.new(self) }
+    end
+  end
 end
 
 module ActiveRecord
   module ConnectionAdapters
     class ConnectionHandler
-      def establish_connection(spec)
-        resolver = ConnectionAdapters::ConnectionSpecification::Resolver.new(Base.configurations)
-        spec = resolver.spec(spec)
+      case "#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}"
+      when '6.1'
+
+        :noop
+
+      when '5.1', '5.2', '6.0'
 
-        owner_to_pool[spec.name] = ActiveRecordHostPool::PoolProxy.new(spec)
+        def establish_connection(spec)
+          resolver = ConnectionAdapters::ConnectionSpecification::Resolver.new(Base.configurations)
+          spec = resolver.spec(spec)
+
+          owner_to_pool[spec.name] = ActiveRecordHostPool::PoolProxy.new(spec)
+        end
+
+      else
+
+        raise "Unsupported version of Rails (v#{ActiveRecord::VERSION::STRING})"
       end
     end
   end
 end
 
 ActiveRecord::ConnectionAdapters::Mysql2Adapter.include(ActiveRecordHostPool::DatabaseSwitch)
+
+if "#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}" == '6.1'
```

Comment:
Could you provide a bit of an explanation around what this patch is doing while the context is fresh? Also, should it be >= 6.1 ? I'm guessing that it'll apply for 7.0 too.